nodes/src/nodes/tool_git/git_repo.py

```
@@ -0,0 +1,972 @@
+        self._tmp: Optional[tempfile.NamedTemporaryFile] = None
+
+    def credentials(
+        self,
+        url: str,
+        username_from_url: Optional[str],
+        allowed_types: int,
+    ) -> pygit2.credentials.Keypair:
+        # Write the private key to a temp file so libgit2 can read it.
+        # The file is cleaned up in close().
+        if self._tmp is None:
+            self._tmp = tempfile.NamedTemporaryFile(mode='w', suffix='.pem', delete=False)
```

Ruff (0.15.12)

[warning] Use a context manager for opening files

(SIM115)